### ISSUE TITLE
Filter by Attribute > Ensure the products displayed on the filter results match the product attributes

### ISSUE THREAD

**Issue by nefeline** [woothemes/woocommerce#42441]
While working on woocommerce/woocommerce#42466 and reviewing woocommerce/woocommerce-blocks#8583, it was possible to verify that the images displayed when filtering products by attributes don't always match the expected results. For example, if you filter by the color blue, products with red color are displayed.

## Steps to Reproduce

1. On a new page, add the Product Collection block and the Product Filters block.
2. On the frontend, filter the products by the color blue
3. Notice how products with the color red are displayed instead:

<img width="457" alt="Screenshot 2023-03-13 at 09 06 03" src="https://user-images.githubusercontent.com/15730971/224697266-f83d0654-251f-4f5f-8da3-8badc11d94f4.png">

This happens because the returned products have the attributes red and blue, but ideally, the displayed products should match the filtered results for consistency and prevent user confusion.

**Comment by gigitux** [comment]
I'm not sure this is an issue, as both products are variable and have `blue` as an option. However, the behavior is reproducible using the new filters block and the Product Collection:

![Image](https://github.com/user-attachments/assets/1cb4a77a-39b5-4ed4-8abd-3d9e37ea7b19)

@roykho, do you think this needs to be addressed? If so, could you update it to reference the new filters block? Otherwise, feel free to close it. Thanks! 🙇

**Comment by roykho** [comment]
@gigitux yes this is still relevant however I will say that this is not actually a bug but more of an enhancement. I will update the issue with more context, thanks for pinging.
